db/version_set_test.cc

```diff
@@ -758,12 +763,29 @@ TEST_F(VersionSetTest, SameColumnFamilyGroupCommit) {
   EXPECT_EQ(kGroupSize - 1, count);
 }
 
-TEST_F(VersionSetTest, HandleValidAtomicGroup) {
+class VersionSetAtomicGroupTest : public VersionSetTestBase,
+                                  public testing::TestWithParam<int> {
+ public:
+  const static int kTestVersionSetRecover;
+  const static int kTestReactiveVersionSetRecover;
+  const static int kTestReactiveVersionSetReadAndApply;
+  VersionSetAtomicGroupTest() : VersionSetTestBase() {}
+
+  void TearDown() override { log_writer_.reset(); }
+
+ protected:
+  std::unique_ptr<log::Writer> log_writer_;
+};
+
+const int VersionSetAtomicGroupTest::kTestVersionSetRecover = 0;
+const int VersionSetAtomicGroupTest::kTestReactiveVersionSetRecover = 1;
+const int VersionSetAtomicGroupTest::kTestReactiveVersionSetReadAndApply = 2;
+
+TEST_P(VersionSetAtomicGroupTest, HandleValidAtomicGroup) {
+  const int test_use_case = GetParam();
```

Comment:
If you use GetParam() to distinguish different test cases, why not make them separate tests?

Reply:
I separate them into different tests.